crates/adapter/src/lib.rs

```diff
@@ -0,0 +1,1429 @@
+use crate::bindings::wasi::clocks::{monotonic_clock, wall_clock};
+use crate::bindings::wasi::io::poll;
+use crate::bindings::wasi::io::streams;
+use crate::bindings::wasi::random::random;
+use core::cell::{Cell, RefCell, RefMut, UnsafeCell};
+use core::ffi::c_void;
+use core::mem::{self, align_of, forget, size_of, MaybeUninit};
+use core::ops::{Deref, DerefMut};
+use core::ptr::{self, null_mut};
+use core::slice;
+use poll::Pollable;
+use wasi::*;
+
+// test
```

Comment:
Probably worth explaining more about the adapter, maybe here or in a README. The fact that a lot of the code is common with `wasi-preview1-component-adapter` from Wasmtime is important, and I'm also curious how it differs beyond the addition of support for Fastly-specific APIs.

Reply:
I put some details about this into `crates/adapter/README.md`.